apps/cowswap-frontend/src/common/pure/PendingOrderNotification/index.tsx

```diff
@@ -28,26 +28,56 @@ const ORDER_TYPE_TITLES: Record<UiOrderType, string> = {
   [UiOrderType.TWAP]: 'TWAP order',
 }
 
-interface PendingOrderNotificationProps {
+export interface PendingOrderNotificationProps {
+  account: string
+  chainId: SupportedChainId
   orderId: string
   kind: OrderKind
   orderType: UiOrderType
   inputAmount: CurrencyAmount<Token>
   outputAmount: CurrencyAmount<Token>
   receiver?: string
   orderCreationHash?: string
+  isSafeWallet: boolean
 }
 
-export function PendingOrderNotification({
-  orderId,
-  kind,
-  orderType,
-  inputAmount,
-  outputAmount,
-  receiver,
-}: PendingOrderNotificationProps) {
-  const { chainId, account } = useWalletInfo()
-  const isSafeWallet = useIsSafeWallet()
+export function getPendingOrderNotificationToast(props: PendingOrderNotificationProps): OnToastMessagePayload | null {
+  const { orderId, kind, inputAmount, outputAmount, receiver, account } = props
+
+  if (!account) return null
+
+  const toAddress = receiver && isAddress(receiver) ? shortenAddress(receiver) : receiver
+
+  const inputAmountElement = formatTokenAmountWithSymbol({
+    amount: inputAmount,
+    tokenSymbol: inputAmount.currency,
+  })
+  const outputAmountElement = formatTokenAmountWithSymbol({
+    amount: outputAmount,
+    tokenSymbol: outputAmount.currency,
+  })
+
+  const baseMessage = (() => {
+    const isSellOrder = kind === OrderKind.SELL
+
+    if (isSellOrder) {
+      return `Sell ${inputAmountElement} for at least ${outputAmountElement}`
+    } else {
+      return `Buy ${inputAmountElement} for at most ${outputAmountElement}`
+    }
+  })()
+
+  return {
+    messageType: ToastMessageType.SWAP_POSTED_API,
+    message: toAddress && receiver ? `${baseMessage}. Receiver: ${toAddress}` : baseMessage,
+    data: {
+      orderUid: orderId,
+    },
+  }
+}
+
+export function PendingOrderNotification(props: PendingOrderNotificationProps) {
```

Comment:
`PendingOrderNotification` receives now all context it needs as props. As now is pure, I also moved it with the pure components